test/tests/performance/pbs_startup_timeout.py

```diff
@@ -0,0 +1,79 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair�s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair�s trademarks, including but not limited to "PBS�",
+# "PBS Professional�", and "PBS Pro�" and Altair�s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.performance import *
+import os
+
+
+class TestStartupTimeout(TestPerformance):
+    """
+    Test that PBS startup does not timeout when there
+    are a very large number of jobs present
+    """
+    def setUp(self):
+        TestPerformance.setUp(self)
+
+    @timeout(500)
+    def test_timeout_in_startup(self):
+        """
+        Test for timeout, i.e. PBS starts successfully after
+        submission of a large number of jobs
+        """
+        # Create a 5MB job script
+        jobscript = "startup_timeout_job.script"
+
+        with open(jobscript, "wb") as js:
+            js.write("#!/bin/sh")
+            js.write("/bin/sleep 20")
+            js.truncate(1024 * 1024 * 5)
+
+        # Submit 1000 jobs
+        job = Job()
+        job.set_execargs(jobscript)
+        for _ in range(1000):
```

Comment:
1000 isn't really a big number, do you think that server will take more than 300 seconds to come up when there are 1000 jobs in db.
I believe this test script will pass with/without your change.
I don't really know how you can test this in a PTL script (maybe you don't need a PTL script for this pull)
Besides "self.server.isUp()" is just a "ps -e" of pbs_server process which will always give you a positive result. you might want to check for "systemd --show-status"